packages/lesswrong/components/ultraFeed/UltraFeedThreadItem.tsx

```diff
@@ -377,8 +405,32 @@ const UltraFeedThreadItem = ({thread, index, settings = DEFAULT_SETTINGS, startR
                 return !!metaInfo?.lastViewed || !!metaInfo?.lastInteracted;
               });
               
+              const allRead = item.hiddenComments.every(h => {
```

Comment:
This addition feels like enough that everything starting from `compressedItems.map` should probably be its own component, or separate function, or something.  If this is all logic that's specific to `UltraFeedCompressedCommentsItem` then maybe it ought to be internalized?  Not sure.